events/importer/kulke.py

```diff
@@ -838,33 +834,34 @@ def simple(field):
             else:
                 setattr(super_event, fieldname, value)
 
-        # The name may vary within a recurring event; hence, take the common part
-        if expand_model_fields(super_event, ["headline"])[0] not in common_fields:
-            words = first_event.headline.split(" ")
-            name = ""
-            while words and all(
-                headline.startswith(name + words[0])
-                for headline in [event.name for event in member_events]
-            ):
-                name += words.pop(0) + " "
-                logger.warning(words)
-                logger.warning(name)
-            if name:
-                super_event.name = name
-            else:
-                # If a common part was not found, default to the first event's name
-                super_event.name = first_event.name
+        # The name may vary within a recurring event; hence, take the common part in each language
+        for lang in self.languages:
+            name_attr = f"name_{lang}"
+            first_name = getattr(first_event, name_attr)
+            words = first_name.split(" ") if first_name else None
 
-        for lang in self.languages.keys():
-            headline = getattr(super_event, "headline_{}".format(lang))
-            secondary_headline = getattr(
-                super_event, "secondary_headline_{}".format(lang)
-            )
-            setattr(
-                super_event,
-                "name_{}".format(lang),
-                make_event_name(headline, secondary_headline),
-            )
+            if name_attr not in common_fields:
+                name = ""
+                member_event_names = [
+                    getattr(event, name_attr) for event in member_events
+                ]
+                while words and all(
+                    member_event_name.startswith(name + words[0])
+                    if member_event_name
+                    else False
+                    for member_event_name in member_event_names
+                ):
+                    name += words.pop(0) + " "
+
+                if name:
+                    setattr(super_event, name_attr, name.rstrip())
+                else:
+                    # If a common part was not found, default to the first event's name
+                    setattr(
+                        super_event,
+                        name_attr,
+                        getattr(first_event, name_attr),
+                    )
```

Comment:
Not necessarily a better way of doing this, but an alternative that's a bit easier to read:
```suggestion
                for word in words:
                    if all(
                        member_event_name and
                        member_event_name.startswith(name + word)
                        for member_event_name in member_event_names
                    ):
                        name += word + " "
                    else:
                        name = name.rstrip()
                        break

                # If a common part was not found, default to the first event's name
                setattr(super_event, name_attr, name or getattr(first_event, name_attr))
```
Assumes that `words` is always a list. Might require some extra formatting from black.